lib/index.js

```diff
@@ -241,12 +245,60 @@ Socket.prototype._read = function (n) {}
  * Socket API
  */
 
-Socket.prototype.bind = function (addr) {
-  return this.bound[addr] = this._protect(nn.Bind( this.binding, addr ));
+Socket.prototype.bind = function (addr, fn) {
+  if (fn) return asyncBindWrapper.call(this, nn.asyncBind);
+  var ret;
+  this.sync('Bind', this, addr, ret);
```

Comment:
> We're not using [ret] after invoking this.sync

yea `this.sync` is a deasync'd `function block()`:
```js
function block (nnMethod, ctx, opt, ret, done) {
  var throttle = setInterval(function(){
    if (ret !== undefined) {
      clearInterval(throttle);
      done();
    }
  }, 10);
  ret = ctx._protect(nn[nnMethod]( ctx.binding, opt ));
  switch (nnMethod) {
    case 'Bind'     : return ctx.bound[opt]       = ret;
    case 'Connect'  : return ctx.connected[opt]   = ret;
  }
}
```

i mean this line in deasync screws that idea up by unfortunately rewriting the function signature:
```js
fn.apply(this, Array.prototype.slice.apply(arguments).concat(cb));
```
to fix that we'll need to maintain the `arguments` object better. `ret` ends up not defined where it needs to be at least `undefined` inside what was `function block()`. I suppose passing `undefined` directly to `this.sync` will do the trick...